modules/default/compliments/compliments.js

```diff
@@ -236,6 +258,25 @@ Module.register("compliments", {
 			compliment.lastElementChild.remove();
 			wrapper.appendChild(compliment);
 		}
+		// if a new set of compliments was loaded from the refresh task
+		// we do this here to make sure no other function is using the compliments list
+		if(this.compliments_new){
+			// use them
+			if(JSON.stringify(this.config.compliments)!== JSON.stringify(this.compliments_new)){
+				// only reset if the contents changes
+				this.config.compliments = this.compliments_new
+				// reset the index
+				this.lastIndexUsed = -1;
+			}
+		}
+		// in test mode only
+		//if (compliments_test_mode) {
```

Comment:
as mentioned above defined but not used here

Reply:
again this was left to help resolve the window.name problem